main.py

```diff
@@ -1,16 +1,147 @@
+import base64
+import datetime
+import glob
+import json
 import os
+import ssl
+import subprocess
+
+import certifi
 import decky_plugin
+from aiohttp import ClientSession
 from settings import SettingsManager
+from typing_extensions import Optional
+
 
 class Plugin:
+    yt_process: Optional[subprocess.Popen] = None
+    music_path = f"{decky_plugin.DECKY_PLUGIN_RUNTIME_DIR}/music"
+    cache_path = f"{decky_plugin.DECKY_PLUGIN_RUNTIME_DIR}/cache"
+    ssl_context = ssl.create_default_context(cafile=certifi.where())
+
     async def _main(self):
-         self.settings = SettingsManager(name="config", settings_directory=decky_plugin.DECKY_PLUGIN_SETTINGS_DIR)
+        self.settings = SettingsManager(
+            name="config", settings_directory=decky_plugin.DECKY_PLUGIN_SETTINGS_DIR
+        )
 
     async def _unload(self):
-        pass
+        if self.yt_process is not None:
+            self.yt_process.terminate()
 
     async def set_setting(self, key, value):
         self.settings.setSetting(key, value)
 
     async def get_setting(self, key, default):
-        return self.settings.getSetting(key, default)
\ No newline at end of file
+        return self.settings.getSetting(key, default)
+
+    async def search_yt(self, term: str):
+        if self.yt_process is not None:
+            self.yt_process.terminate()
+        self.yt_process = subprocess.Popen(
```

Comment:
Given the async context of the plugin, it would be better to use [`asyncio.create_subprocess_exec`](https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.create_subprocess_exec).

Reply:
Agreed, fixed in 49d32b6.